inst/stan/data/delays.stan

```diff
@@ -1,14 +1,13 @@
-  int delays;                  // no. of delay distributions
-  int n_uncertain_mean_delays;   // no. of delay distributions with uncertain mean
-  int n_uncertain_sd_delays;     // no. of delay distributions with uncertain sd
-  int n_fixed_delays;     // no. of delay distributions with uncertain sd
-  // indices of delay distributions with uncertainty
-  int<lower = 1, upper = delays> uncertain_mean_delays[n_uncertain_mean_delays];
-  int<lower = 1, upper = delays> uncertain_sd_delays[n_uncertain_sd_delays];
-  int<lower = 1, upper = delays> fixed_delays[n_fixed_delays];
-  real delay_mean_sd[delays];  // prior sd of mean incubation period
-  real delay_mean_mean[delays];// prior mean of mean incubation period
-  real delay_sd_mean[delays];  // prior sd of sd of incubation period
-  real delay_sd_sd[delays];    // prior sd of sd of incubation period
-  int<lower = 1> delay_max[delays];       // maximum incubation period
-  int delay_dist[delays];       // 0 = lognormal; 1 = gamma
+  int<lower = 0> delay_n;                  // number of delay distribution distributions
+  int<lower = 0> delay_n_p;                // number of parametric delay distributions
+  int<lower = 0> delay_n_np;                // number of nonparametric delay distributions
+  real delay_mean_mean[delay_n_p]; // prior mean of mean delay distribution
+  real<lower = 0> delay_mean_sd[delay_n_p];   // prior sd of mean delay distribution
```

Comment:
rather than differentiating between delays and gts we could just pass them all together and then have indices that pull out the ones we need for any given step. This would make things a bit shorter and more concise + prepare the ground for fitting to multiple observations. 

It would mean that the generation time wasn't present as a clear parameter in the model output (with this also being true for the other delays) and this might make development harder and diagnosing issues with the model. That could be dealt with by having better post-processing which we would also need if we were going to have multiple observations.

Reply:
Same point about truncation distributions etc..

Reply:
Doing this would require changing how pre-processing works and the design of that could be so painful as to not make doing this worthwhile.